lib/EnsEMBL/REST/Model/GenomicAlignment.pm

```diff
@@ -76,18 +71,24 @@ sub get_alignment {
 
   #Get the compara DBAdaptor
   $c->forward('get_adaptors');
+
+  #Get method
+  my $method_type = $c->request->parameters->{method} || 'EPO';
+  my $method = $c->stash->{method_adaptor}->fetch_by_type($method_type);
```

Comment:
It doesn't seem like $method gets used for anything of substance, now you've renamed everything to method_type. Is that intentional?

Reply:
`$method` is only used to get the "class" and nothing else.
I was actually wondering about that. To be as close as possible to the compara API, `method_type` should a string and `method` the object. But by doing that I indeed had to rename the variable in a few more places, which admittedly clutters the diff. Alternatively, to make the diff clearer, I could leave `$method` as it was and name the new variable `$method_object` ?

Reply:
Tell you what, move the invocation of $method->class to a more up-front test rather than a postfix so it's more obvious. The line is too long for github anyway. Is $method->class the same or different to $method->isa?

Reply:
`class` is one of the fields _we_ define in the object, cf https://github.com/Ensembl/ensembl-compara/blob/release/94/modules/Bio/EnsEMBL/Compara/Method.pm#L138-L155 http://www.ensembl.org/info/docs/api/compara/compara_schema.html#method_link
It's a string that we use to tell what type of data the pipeline has produced.

Reply:
Are you content to leave this as is?